### PR TITLE
UX improvements

### PR DESCRIPTION
**Pull request by alex-miller-0**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -108,11 +123,23 @@ class LatticeKeyring extends EventEmitter {
...
           // Add these indices
           addrs.forEach((addr, i) => {
-            if (this.accounts.indexOf(addr) === -1) {
-              this.accounts.push(addr)
-              this.accountIndices.push(this.unlockedAccount+i)
+            let alreadySaved = false;
+            for (let j = 0; j < this.accounts.length; j++) {
+              if ((this.accounts[j] === addr) && 
+                  (this.accountOpts.walletUID[j] === walletUID) &&
+                  (this.accountOpts.hdPath === this.hdPath))
```
Comment (alex-miller-0) [inline index.js:133]: Not sure what happened here. These should both be `this.accountOpts[j].<param>`

File: index.js [hunk 2]
```diff
@@ -243,19 +278,26 @@ class LatticeKeyring extends EventEmitter {
   signMessage(address, msg) {
     return new Promise((resolve, reject) => {
       this._unlockAndFindAccount(address)
-      .then((addrIdx) => {
+      .then((accountIdx) => {
+        // Make sure the account is associated with the current wallet
+        if (this.accountOpts[accountIdx].walletUID !== this._getCurrentWalletUID()) {
+          return reject(new Error('Account associated with different wallet. ' +
```
Comment (alex-miller-0) [inline index.js:284]: I like the language above better. Guess it didn't get copied to this one.

```
          return reject(new Error('Account on a different wallet. ' +
                                  'Please switch to the correct wallet on your Lattice.'));
                                  ```

File: index.js [hunk 3]
```diff
@@ -243,19 +278,26 @@ class LatticeKeyring extends EventEmitter {
   signMessage(address, msg) {
     return new Promise((resolve, reject) => {
       this._unlockAndFindAccount(address)
```
Comment (KevinVitale) [inline index.js:280]: In both calls to `_unlockAndFindAccount`, the initial logic within subsequent `then` is the same. Should that initial logic be merged into the `Promise` that gets returned?

Comment (KevinVitale) [reply index.js:280]: Specifically:
```
         // Make sure the account is associated with the current wallet
         if (this.accountOpts[accountIdx].walletUID !== this._getCurrentWalletUID()) {
           return reject(new Error('Account associated with different wallet. ' +
                                   'Please switch to this wallet on your Lattice to continue.'));
         }
```

Comment (alex-miller-0) [reply index.js:275]: 5666dd5

File: index.js (hunk 1 above)
Comment (alex-miller-0) [reply index.js]: 5666dd5

**Pull request merged**